stac_api/api/extensions/tiles_search.py

```diff
@@ -0,0 +1,235 @@
+            # postgres client to self.get_assets
+            # with self.reader(None, item=asset, **self.reader_options) as src_dst:
+            with self.reader(None, item=asset) as src_dst:
+                return src_dst.tile(x, y, z, **kwargs)
+
+        return mosaic_reader(mosaic_assets, _reader, x, y, z, **kwargs)
+
+
+@dataclass
+class AssetParams(DefaultDependency):
+    """Band names and Expression parameters."""
+
+    assets: Optional[str] = Query(
+        None,
+        title="Asset names",
+        description="comma-delimited asset names.",
+    )
+
+    def __post_init__(self):
+        """Post Init."""
+        if self.assets is not None:
+            self.kwargs["assets"] = self.assets.split(",")
+
+
+@dataclass
+class DatasetDependency(DefaultDependency):
+    """Add STAC search params to query."""
+
+    collections: Optional[str] = Query(
+        None, title="Collections", description="Collections to search within"
+    )
+
+    ids: Optional[str] = Query(None)
+    datetime: Optional[str] = Query(None)
+    limit: Optional[int] = Query(10)
+    query: Optional[str] = Query(None)
+    token: Optional[str] = Query(None)
+    fields: Optional[str] = Query(None)
+    sortby: Optional[str] = Query(None)
+
+    def __post_init__(self):
+        """Post Init."""
+        if self.collections:
+            self.kwargs["collections"] = self.collections.split(",")
+        if self.ids:
+            self.kwargs["ids"] = self.ids.split(",")
+        if self.datetime:
+            self.kwargs["datetime"] = self.datetime
+        if self.limit:
+            self.kwargs["limit"] = self.limit
+        if self.query:
+            self.kwargs["query"] = self.query
+        if self.token:
+            self.kwargs["token"] = self.token
+        if self.fields:
+            self.kwargs["fields"] = self.fields.split(",")
+        if self.sortby:
+            self.kwargs["sortby"] = self.sortby.split(",")
+
+
+@attr.s
+class TilesSearchExtension(ApiExtension):
+    """Custom Titiler Extension.
+
+    Custom extension for sending query results to a titiler DynamicSTAC endpoint
+    """
+
+    client: CoreCrudClient = attr.ib(default=attr.Factory(CoreCrudClient))
+
+    def register(self, app: FastAPI) -> None:
+        """Register the extension with a FastAPI application.
+
+        Args:
+            app: target FastAPI application.
+
+        Returns:
+            None
+        """
+        dynamic_stac_endpoint = MosaicTilerFactory(
+            reader=DynamicStacBackend,
+            reader_options={"client": self.client},
+            # path_dependency=MosaicParams,
+            dataset_reader=STACReader,
+            dataset_dependency=DatasetDependency,
+            layer_dependency=AssetParams,
+            router_prefix="test-dynamic-search",
+        )
+
+        app.include_router(
+            dynamic_stac_endpoint.router, prefix="/dynamic-stac", tags=["Dynamic STAC"]
```

Comment:
It seems unnecessary to have both of these router prefixes, and it looks like `/dynamic-stac` overrides `test-dynamic-search` anyways. Also, open to other prefix names.

Reply:
it is unnecessary, I think i'd rather include the prefix on the call to `include_router`.